src/plumbing.rs

```diff
@@ -237,3 +236,49 @@ where
         S: Eq + Hash,
         Q::Key: Borrow<S>;
 }
+
+/// The error returned when a query could not be resolved due to a cycle
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub struct CycleError<K> {
+    /// The queries that were part of the cycle
+    pub(crate) cycle: Vec<K>,
+    pub(crate) changed_at: Revision,
+    pub(crate) durability: Durability,
```

Comment:
Might make sense to add comments explaining the semantics of these two fields.

Reply:
Actually, now that you mention it, I don't think we need this type in the final version, so I've removed it.